langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/FromJsonWithType.java

```diff
@@ -94,6 +85,15 @@ private static Object convert(Object value, Type targetType, List<TypeValuePair>
                     BLangExceptionHelper.getErrorMessage(RuntimeErrors.CANNOT_CONVERT_NIL, targetType));
```

Comment:
Since we didn't get to this point previously, there's probably no test when the value is null but the target type does not allow nil. Shall we add a test for that too?

Reply:
Fixed in 840a2b3